services/datamanager/builtin/builtin_capture_test.go

```diff
@@ -245,63 +251,53 @@ func TestSwitchResource(t *testing.T) {
 	err = b.Reconfigure(context.Background(), deps2, config)
 	test.That(t, err, test.ShouldBeNil)
 
-	dataBeforeSwitch, err := getSensorData(captureDir)
-	test.That(t, err, test.ShouldBeNil)
-
-	// Test that sensor data is captured from the new collector.
+	// wait for all the files on disk to
 	waitForCaptureFilesToExceedNFiles(captureDir, len(getAllFileInfos(captureDir)), logger)
 	testFilesContainSensorData(t, captureDir)
 
-	filePaths := getAllFilePaths(captureDir)
-	test.That(t, len(filePaths), test.ShouldEqual, 2)
+	// Test that sensor data is captured from the new collector.
+	var (
+		captureDataHasZeroReadings    bool
+		captureDataHasNonZeroReadings bool
+	)
+
+	for _, fp := range getAllFilePaths(captureDir) {
```

Comment:
I updated the test from needing to care about how many capture files are written (which we can't control) and to instead simply assert that we have capture readings from both the pre and post `Reconfigure()` "arm1" instances.